### PR TITLE
Feature remove

### PR DESCRIPTION
**Pull request by fortuneonyeka**
In this milestone, I
 Create a new JavaScript file for the new functionality.
 Implement a function for adding a new task (add a new element to the array).
 Implement a function for editing task descriptions.
 Implement a function for deleting a task (remove an element from the array).
 Implement a function for the "Clear all completed" button (use filter() method).
 By default, new tasks should have the property completed set to false and the property index set to the value of the new array length.
 Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.
 All changes to the To-Do List saves be saved in local storage.

### CODE REVIEW

File: src/task.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    };
+    buttonsDiv.appendChild(buttonMenu);
+  });
+}
+
+function updateLocalStorage() {
+  localStorage.toDoList = JSON.stringify(toDoList);
+  document.location.reload();
+  return false;
+}
```
Comment (de-scholar) [inline src/task.js:87]: - [ ] This update function doesn't work well, when I update a given task, and refresh the page, the changes disappear, would you address that issue?

File: src/task.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+function updateIndex() {
+  let counter = 0;
+  toDoList.forEach((todo) => {
+    todo.index = counter;
+    counter++;
+  });
+  updateLocalStorage();
+}
```
Comment (de-scholar) [inline src/task.js:96]: - [ ] The Update index also doesn't work properly. Actually, when I delete a given task, all of indexes should be updated and reorder, in your case it's not that behavior
Screenshot from 2021-08-21 15-18-49

File: src/task.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  document
+    .querySelector(`.buttons-container-${index}`)
+    .classList.toggle('hidden');
+}
+
+function clearCompleted() {
+  toDoList = JSON.parse(localStorage.toDoList);
+  toDoList = toDoList.filter((todo) => todo.completed === false);
+  updateLocalStorage();
+}
```
Comment (de-scholar) [inline src/task.js:113]: - [ ] When a task is empty, and marked as complete, this function doesn't remove it, would you fix that as well?

**Review by de-scholar — CHANGES_REQUESTED** [review]
Hi @fortuneonyeka , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by descholar-ceo — APPROVED** [review]
Hi @fortuneonyeka , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**